### PR TITLE
ci, build: update CI, libraries, python dependencies, and format

### LINKED ISSUE #84: `tensorboards-web-app` fails to build during integration tests
```
 RuntimeError: Failed to build charm .:
Packing the charm.
Launching environment to pack for base name='ubuntu' channel='20.04' architectures=['amd64'] (may take a while the first time but it's reusable)
Packing the charm
Packing the charm.
Building charm in '/root'
Running step PULL for part 'charm'
Running step BUILD for part 'charm'
Parts processing error: Failed to run the build script for part 'charm'.
Failed to build charm for bases index '0'.
Full execution log: '/home/runner/.local/state/charmcraft/log/charmcraft-20230724-093116.474552.log'
```

For details and logs, see the [latest integration test job](https://github.com/canonical/kubeflow-tensorboards-operator/actions/runs/5642971290/job/15283883485#step:4:522) for #83

### PR DESCRIPTION
**Pull request by DnPlas**
This commit updates the CI with standard CI configuration (juju, actions, deps) that is shared across multiple repositories. All python dependencies are also updated to their latest working versions. This PR also includes library updates. Finally, some formatting was applied on non-compliant files (copyright, style).

This change also contains a fix for #84, an issue caused by a missing toolchain in the charm that allows building certain python packages. For more information, please refer to canonical/bundle-kubeflow#648.

Fixes #84 
Part of canonical/bundle-kubeflow#648

### CODE REVIEW

File: charms/tensorboard-controller/requirements-integration.in
Comment (phoevos) [inline charms/tensorboard-controller/requirements-integration.in:1]: I think we should also explicitly add `pytest`.

File: charms/tensorboard-controller/requirements-integration.in [hunk 1]
```diff
@@ -1,7 +1,8 @@
 # Pinning to <3.0 to ensure compatibility with the 2.9 controller version
 # Note: 3.0 is not being maintained anymore
 juju<3.0
+lightkube
 pytest-operator
+pyyaml
 tenacity
 websockets
```
Comment (phoevos) [inline charms/tensorboard-controller/requirements-integration.in:8]: This was my addition, but I can't really tell why it's needed now 😂

File: charms/tensorboards-web-app/requirements-integration.in
Comment (phoevos) [inline charms/tensorboards-web-app/requirements-integration.in:1]: Should we also explicitly add `pytest` here?

**Review by phoevos — CHANGES_REQUESTED** [review]
Please don't forget the following:
```diff
diff --git a/charms/tensorboard-controller/requirements-lint.in b/charms/tensorboard-controller/requirements-lint.in
index c77cd87..07a4a51 100644
--- a/charms/tensorboard-controller/requirements-lint.in
+++ b/charms/tensorboard-controller/requirements-lint.in
@@ -1,9 +1,8 @@
+black
 codespell
-flake8<6
+flake8
 flake8-builtins
-# Pinned because `flake8-copyright==0.2.3` is incompatible with `flake8>=6`.  Can unpin this
-# when https://github.com/savoirfairelinux/flake8-copyright/pull/20 or a similar fix is released
 flake8-copyright
+isort
 pep8-naming
 pyproject-flake8
--r requirements-fmt.txt
diff --git a/requirements-integration.in b/requirements-integration.in
index a07d242..e1a657e 100644
--- a/requirements-integration.in
+++ b/requirements-integration.in
@@ -1,4 +1,6 @@
 # Pinning to <3.0 to ensure compatibility with the 2.9 controller version
 # Note: 3.0 is not being maintained anymore
 juju<3.0
+pytest
 pytest-operator
+pyyaml
```

File: charms/tensorboard-controller/requirements-integration.in (hunk 1 above)
Comment (DnPlas) [reply charms/tensorboard-controller/requirements-integration.in:8]: Well, it is imported and used in `tensorboard-controller/tests/integration/test_charm.py`, what do you mean you don't know?

Comment (phoevos) [reply charms/tensorboard-controller/requirements-integration.in:9]: Are you talking about `websockets`?

File: charms/tensorboard-controller/requirements-lint.in
Comment (phoevos) [inline charms/tensorboard-controller/requirements-lint.in:1]: `black` is missing here!

File: charms/tensorboard-controller/requirements-integration.in (hunk 1 above)
Comment (DnPlas) [reply charms/tensorboard-controller/requirements-integration.in:9]: Oh, shite, I thought of pyyaml (as is the shown change), sorry, we should've clarified from the beginning. I can remove it if it's not needed.

**Review by phoevos — APPROVED** [review]

**Pull request merged**